swap_meet/vendor.py

```diff
@@ -1,2 +1,207 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        '''
+        Constructor of Vendor class:
+        Parameters:
+            inventory - represents all of the objects of the classs Item
+              that self has and is willing to swap with other vendors
+        '''
+
+        #If no argument is passed for inventory it will initialize as an empty list
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, new_item):
+
+        self.inventory.append(new_item)
+        return new_item
+    
+    def remove(self, old_item):
+        '''
+        input:
+            self -
+            old_item - object from the Item class that self will remove from its inventory list 
+        output:
+            old_item - if old_item is found in the inventory list it will be removed and returned 
+                side effect- inventory will have the old_item removed 
+            False - if old_item is not found inventory will not change
+        '''
+
+        if old_item in self.inventory:
+
+            self.inventory.remove(old_item)
+            return old_item
+
+        else:
+            return False
+        
+    
+    def get_by_id(self, items_id):
+        '''
+        Finds the item in self.inventory with the id = items_id
+        '''
+
+        for item in self.inventory:
+            if items_id == item.id: 
+                return item
+    
+
+    
+    def swap_items(self, other_vendor, my_item, others_item):
+        '''
+    input: 
+        self -
+        other_vendor - object of the same class vendor that self wants to trade with, 
+        my_item - Item self wants to swap, 
+        others_item- item other_vendor wants to swap
+    output: 
+        True - if self and othr_vendor have items that the other wants to swap
+            Side effect- the inventory parameter for self and other_vendor is modified by having the new item and
+                removing the old item
+        False - if self or other_vendor does not have its corrsponding item in their inventory parameter
+        '''
+
+        #Check each of the vendors has the correct item to swap 
+
+        if my_item in self.inventory and others_item in other_vendor.inventory:
```

Comment:
[suggestion] I like inverting the logic of the condition to be able to treat it like a guard clause, which allows us to outdent the main logic of the function something like the following:

```python
        if my_item not in self.inventory or others_item not in other_vendor.inventory:
            return False

        # Everything else is now less indented
```

Note that this uses [De Morgan's Law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws). Basically, `x and y` is equal to `not x or not y`